lib/pure/algorithm.nim

```diff
@@ -31,28 +52,32 @@ template fillImpl[T](a: var openArray[T], first, last: int, value: T) =
     inc(x)
 
 proc fill*[T](a: var openArray[T], first, last: Natural, value: T) =
-  ## fills the slice ``a[first..last]`` with ``value``.
+  ## Fills the slice ``a[first..last]`` with ``value``.
```

Comment:
What happens if I do:
```nim
import algorithm

var a: array[5, int]
a.fill(3, 10, 9)
```

One should not need to guess/try stuff like that.

If this works, then it should be shown in the example (see for example [math.log](https://nim-lang.github.io/Nim/math.html#log%2CT%2CT), where I was curious to see what happens if I give `x<=0` to `log`. 
If it raises an error, it should be mentioned in the docs (Something along the lines: "If an invalid range is passed, it raises `IndexError`"), and you can then write an example like this:

```nim
var a: array[5, int]
doAssertRaises(IndexError, a.fill(3, 10, 9))
```